src/redux/reducers/charts/index.js

```diff
@@ -0,0 +1,141 @@
+                borderColor: '',
+                backgroundColor: '',
+                type: 'line',
+                fill: true,
+            }],
+        },
+        barCharts: {
+            title: '',
+            type: 'line',
+            labels: [],
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: '',
+            },
+            {
+                title: '',
+                values: [],
+                backgroundColor: '',
+            },
+            {
+                title: '',
+                values: [],
+                backgroundColor: '',
+            },
+            {
+                title: '',
+                values: [],
+                backgroundColor: '',
+            }],
+        },
+        horizontalBarCharts: {
+            title: '',
+            type: 'line',
+            labels: [],
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: '',
+            },
+            {
+                title: '',
+                values: [],
+                backgroundColor: '',
+            }],
+        },
+        radarCharts: {
+            title: '',
+            type: 'line',
+            labels: [],
+            legendPosition: 'bottom',
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: '',
+                fill: true,
+            },
+            {
+                title: '',
+                values: [],
+                borderColor: '',
+                backgroundColor: '',
+                fill: true,
+            }],
+        },
+        polarAreaCharts: {
+            title: '',
+            type: 'line',
+            labels: [''],
+            legendPosition: 'bottom',
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: [],
+            }],
+        },
+        pieCharts: {
+            title: '',
+            type: 'line',
+            labels: [],
+            legendPosition: 'bottom',
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: [],
+            }],
+        },
+        doughnutCharts: {
+            title: '',
+            type: 'line',
+            labels: [],
+            dataSet: [{
+                title: '',
+                values: [],
+                backgroundColor: [],
+            }],
+        },
+    },
+    isLoading: false,
+    isFirstTime: true,
+});
+
+export default function charts(state = initialState, action) {
+    switch (action.type) {
+        case LOADING_CHARTS_DATA:
+            return { ...state, isLoading: true };
+        case LOAD_CHARTS_DATA:
+            return Object.assign({}, state, {
+                ...state,
+                isLoading: false,
+                isFirstTime: false,
+                data: action.payload.chartsData,
+            });
```

Comment:
same ^^
also here you only need to use one Object.assign or ...state not both, so you can:
```
return {
    ...state,
    data,
};
```